packages/back-end/src/jobs/informationSchema.ts

```diff
@@ -0,0 +1,64 @@
+import Agenda, { Job } from "agenda";
+import { DataSourceInterface } from "../../types/datasource";
+import { initializeDatasourceInformationSchema } from "../services/datasource";
+import { logger } from "../util/logger";
+import { updateDataSource } from "../models/DataSourceModel";
+
+const CREATE_INFORMATION_SCHEMA_JOB_NAME = "createInformationSchema";
+type CreateInformationSchemaJob = Job<{
+  datasource: DataSourceInterface;
```

Comment:
This should just be a string id, not the full interface object.  Jobs run asynchronously, so when the job runs you always want to make sure you're using the most up-to-date version in Mongo and not whatever existing at the time the job was queued.  Otherwise, you risk overwriting recent changes.

Also, you can think of the attributes here as the primary key of the job.  The same data source should always have the same primary key, even if the fields inside the data source change.  Otherwise, you risk having duplicate jobs for the same data source running.